### DISCUSSION TITLE
Find the internal html

### DISCUSSION THREAD

**Discussion by gavinpotter** [writer/writer-framework#85] [Q&A]
A very nice way of producing quick backends without having to go into html or css.  However, there are a few things that do require css classes to work.  I understand that the css is only applied to the outer html in each component.  What I can't figure out is how to identify what the inner html is, so that I can apply CSS to that.  So, for a concrete example, I want to make all my buttons have the same width irrespective of the size of the text within them.  Any guidance on how to identify the html that controls the button width and how to create a css class that changes it would be very gratefully received.

Thanks

**Comment by ramedina86** [comment]
You're right that many things require CSS. There's a trade off between simplicity and flexibility and I'm still trying to navigate that, but, unlike some other frameworks, at least with the option of CSS. Width (of things in general) has come up quite often so it'll likely end up becoming a standard setting.

You have two options, you can look at the source code of the components, or perhaps more easily use _Inspect element_ in your browser.

The _Button_ component has a `div` and a `button` within it. In the next release, the `button` will become the root element.

You can write a style such as:

```css
.CoreButton button, button.CoreButton {
width: 200px;
}
```
